compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -157,8 +160,31 @@ impl<'a, 'tcx> TerminatorCodegenHelper<'tcx> {
         destination: Option<(ReturnDest<'tcx, Bx::Value>, mir::BasicBlock)>,
         mut unwind: mir::UnwindAction,
         copied_constant_arguments: &[PlaceRef<'tcx, <Bx as BackendTypes>::Value>],
+        instance: Option<Instance<'tcx>>,
         mergeable_succ: bool,
     ) -> MergingSucc {
+        let tcx = bx.tcx();
+        if let Some(instance) = instance {
+            let def_id = instance.def_id();
+            if !def_id.is_local()
+                && tcx.is_compiler_builtins(LOCAL_CRATE)
+                && !should_codegen_locally(tcx, &instance)
+            {
+                if destination.is_some() {
+                    let callee = with_no_trimmed_paths!(tcx.def_path_str(def_id));
+                    tcx.dcx().emit_err(CompilerBuiltinsCannotCall { callee });
+                } else {
+                    info!(
+                        "compiler_builtins call to diverging function {:?} replaced with abort",
+                        instance.def_id()
+                    );
+                    bx.abort();
+                    bx.unreachable();
+                    return MergingSucc::False;
+                }
+            }
+        }
```

Comment:
cg_clif will need this change too.

Reply:
If I apply this change to cranelift, how would I verify that it works?

Reply:
https://github.com/rust-lang/rustc_codegen_cranelift/#building-and-testing-with-changes-in-rustc-code has instructions.

Reply:
And the new code would likely be somewhere around https://github.com/rust-lang/rustc_codegen_cranelift/blob/4cf4ffc6ba514f171b3f52d1c731063e4fc45be3/src/abi/mod.rs#L374